### ISSUE TITLE
Imosible execute "make flash monitor" following OTA-ESP32 guide

### ISSUE THREAD

**Issue by voluta-coop** [thingsboard/esp32-ota#5]
Hi! I'm following this guide https://thingsboard.io/docs/samples/esp32/ota/ but when I execute "make flash monitor" I get these errors:

"In file included from /home/admin1/esp/esp32-ota/main/main.c:10:
/home/admin1/esp/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp]
#warning "esp_event_loop.h is deprecated, please include esp_event.h instead"
^~~~~~~
/home/admin1/esp/esp32-ota/main/main.c: In function 'mqtt_event_handler':
/home/admin1/esp/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_ANY' not handled in switch [-Werror=switch]
switch (event->event_id)
^~~~~~
/home/admin1/esp/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_DELETED' not handled in switch [-Werror=switch]
cc1: some warnings being treated as errors
make[1]: *** [/home/admin1/esp/esp-idf/make/component_wrapper.mk:292: main.o] Error 1
make: *** [/home/admin1/esp/esp-idf/make/project.mk:635: component-main-build] Error 2".

I'd like to know how I can solve it.

Thank you very much, I'm learning a lot with thingsboard!!!!

**Comment by mr-engin3er** [comment]
Same problem when I execute "make flash monitor".
I followed every instruction from https://thingsboard.io/docs/samples/esp32/ota/
ESP-IDF is successfully installed and working as I flashed hello-world project from esp-idf examples successfully.

```
In file included from /home/dheeraj/Development/thingsboard/esp32-ota/main/main.c:10:
/home/dheeraj/Development/esp/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp]
 #warning "esp_event_loop.h is deprecated, please include esp_event.h instead"
  ^~~~~~~
/home/dheeraj/Development/thingsboard/esp32-ota/main/main.c: In function 'mqtt_event_handler':
/home/dheeraj/Development/thingsboard/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_ANY' not handled in switch [-Werror=switch]
     switch (event->event_id)
     ^~~~~~
/home/dheeraj/Development/thingsboard/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_DELETED' not handled in switch [-Werror=switch]
cc1: some warnings being treated as errors
make[1]: *** [/home/dheeraj/Development/esp/esp-idf/make/component_wrapper.mk:292: main.o] Error 1
make: *** [/home/dheeraj/Development/esp/esp-idf/make/project.mk:642: component-main-build] Error 2

```

**Comment by marianoceneri** [comment]
> Hi! I'm following this guide https://thingsboard.io/docs/samples/esp32/ota/ but when I execute "make flash monitor" I get these errors:
> 
> "In file included from /home/admin1/esp/esp32-ota/main/main.c:10:
> /home/admin1/esp/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp]
> #warning "esp_event_loop.h is deprecated, please include esp_event.h instead"
> ^~~~~~~
> /home/admin1/esp/esp32-ota/main/main.c: In function 'mqtt_event_handler':
> /home/admin1/esp/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_ANY' not handled in switch [-Werror=switch]
> switch (event->event_id)
> ^~~~~~
> /home/admin1/esp/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_DELETED' not handled in switch [-Werror=switch]
> cc1: some warnings being treated as errors
> make[1]: *** [/home/admin1/esp/esp-idf/make/component_wrapper.mk:292: main.o] Error 1
> make: *** [/home/admin1/esp/esp-idf/make/project.mk:635: component-main-build] Error 2".
> 
> I'd like to know how I can solve it.
> 
> Thank you very much, I'm learning a lot with thingsboard!!!!

Same problem here ... is this platform alive?

**Comment by arttash** [comment]
> Hi! I'm following this guide https://thingsboard.io/docs/samples/esp32/ota/ but when I execute "make flash monitor" I get these errors:
> 
> "In file included from /home/admin1/esp/esp32-ota/main/main.c:10: /home/admin1/esp/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp] #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" ^~~~~~~ /home/admin1/esp/esp32-ota/main/main.c: In function 'mqtt_event_handler': /home/admin1/esp/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_ANY' not handled in switch [-Werror=switch] switch (event->event_id) ^~~~~~ /home/admin1/esp/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_DELETED' not handled in switch [-Werror=switch] cc1: some warnings being treated as errors make[1]: *** [/home/admin1/esp/esp-idf/make/component_wrapper.mk:292: main.o] Error 1 make: *** [/home/admin1/esp/esp-idf/make/project.mk:635: component-main-build] Error 2".
> 
> I'd like to know how I can solve it.
> 
> Thank you very much, I'm learning a lot with thingsboard!!!!

Yeah I am having the same issue currently. Were you able to solve this issue?

> Hi! I'm following this guide https://thingsboard.io/docs/samples/esp32/ota/ but when I execute "make flash monitor" I get these errors:
> 
> "In file included from /home/admin1/esp/esp32-ota/main/main.c:10: /home/admin1/esp/esp-idf/components/esp_event/include/esp_event_loop.h:2:2: warning: #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" [-Wcpp] #warning "esp_event_loop.h is deprecated, please include esp_event.h instead" ^~~~~~~ /home/admin1/esp/esp32-ota/main/main.c: In function 'mqtt_event_handler': /home/admin1/esp/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_ANY' not handled in switch [-Werror=switch] switch (event->event_id) ^~~~~~ /home/admin1/esp/esp32-ota/main/main.c:76:5: error: enumeration value 'MQTT_EVENT_DELETED' not handled in switch [-Werror=switch] cc1: some warnings being treated as errors make[1]: *** [/home/admin1/esp/esp-idf/make/component_wrapper.mk:292: main.o] Error 1 make: *** [/home/admin1/esp/esp-idf/make/project.mk:635: component-main-build] Error 2".
> 
> I'd like to know how I can solve it.
> 
> Thank you very much, I'm learning a lot with thingsboard!!!!

Yeah I am currently facing the same issue. Were you able to solve this problem?

**Comment by kmilokeepex** [comment]
Same problem here ... some one have a solution?

**Comment by iwas108** [comment]
Try ```idf.py build``` to check if the code has any error. If the build process is ok, continue to ```idf.py -p /dev/ttyACM0 flash monitor``` to flash and monitor.
